File: php/Kode program/Mobil.php
<?php
require_once "Kendaraan.php";

class Mobil extends Kendaraan {
    private $doors;
    private $fuelType;

    public function __construct($brand, $model, $year, $price, $doors, $fuelType) {
        parent::__construct($brand, $model, $year, $price);
        $this->doors = $doors;
        $this->fuelType = $fuelType;
    }

    public function getDoors() {
        return $this->doors;
    }

    public function getFuelType() {
        return $this->fuelType;
    }
}
?>

File: cpp/code program/mobil.cpp
#pragma once
#include <iostream>
#include <string>
#include "kendaraan.cpp"

using namespace std;

class Mobil : public Kendaraan
{
private:
    int doors;
    string fuelType;

public:
    Mobil()
    {

    }

    Mobil(string b, string m, int y, double p, int d, string f): Kendaraan(b, m, y, p)
    {
        this->doors = d;
        this->fuelType = f;
    }

    void setJumlahpintu(int doors)
    {
        this->doors = doors;
    }


    void setfuelType(string fuelType)
    {
        this->fuelType = fuelType;
    }

    int getdoors()
    {
        return this->doors;
    }

    string getfuelType()
    {
        return this->fuelType;
    }

    ~Mobil()
    {

    }
};
